### PR TITLE
🐛 Log output when onError set

### PR DESCRIPTION
**Pull request by rcebulko**
A change in https://github.com/ampproject/amphtml/pull/34105 accidentally made all type-checks use a custom logger that swallowed output, when we only needed to take a copy of the log output. This patches that issue, unmasking typecheck errors.

### CODE REVIEW

File: build-system/tasks/check-types.js [hunk 1]
```diff
@@ -283,13 +283,20 @@ async function typeCheck(targetName) {
     return;
   }
 
+  if (target.onError) {
+    // If an onError handler is defined, steal the output and log it manually.
+    opts.logger = (m) => {
+      errorMsg = m;
+      log(m);
```
Comment (samouri) [inline build-system/tasks/check-types.js:290]: We probably should not be calling `log(m)` here. The `onError` fn should be in charge of determining output (we want it swallowed).

File: build-system/tasks/check-types.js [hunk 2]
```diff
@@ -283,13 +283,20 @@ async function typeCheck(targetName) {
...
+    };
+  }
+
   let errorMsg;
```
Comment (samouri) [inline build-system/tasks/check-types.js:294]: Doesn't this var declaration need to be above the if statement?

File: build-system/tasks/check-types.js (hunk 1 above)
Comment (rcebulko) [reply build-system/tasks/check-types.js]: You're right, removed.

File: build-system/tasks/check-types.js (hunk 2 above)
Comment (rcebulko) [reply build-system/tasks/check-types.js]: Need, no (hoisted); should, definitely. Moved.

File: build-system/tasks/check-types.js [hunk 3]
```diff
@@ -284,12 +284,16 @@ async function typeCheck(targetName) {
   }
 
   let errorMsg;
+  if (target.onError) {
+    // If an onError handler is defined, steal the output and log it manually.
```
Comment (rcebulko) [inline build-system/tasks/check-types.js:288]:
```suggestion
    // If an onError handler is defined, steal the output and let onError handle
    // logging
```

**Review by samouri — APPROVED** [review]

**Review by kristoferbaxter — APPROVED** [review]

**Pull request merged**